### PR TITLE
feat(parser): Recover from more invalid syntax in the parser

### PR DESCRIPTION
**Pull request by Marwes**

### CODE REVIEW

File: parser/src/grammar.lalrpop [hunk 1]
```diff
@@ -75,6 +75,11 @@ IdentStr: &'input str = {
     "(" <"Operator"> ")" => <>,
 };
 
+SkipExtraTokens: () = {
+    => (),
```
Comment (brendanzab) [inline parser/src/grammar.lalrpop]: Heh - this is interesting syntax @nikomatsakis! 😆

**Comment by brendanzab** [comment]
Would be nice to get some test coverage on the errors we are emitting from the compiler...

**Comment by Marwes** [comment]
> Would be nice to get some test coverage on the errors we are emitting from the compiler...

Yep, it is hard to come up with examples with poor error messages though. It is only really after this change that cascading errors end up being a problem at least.

File: parser/src/grammar.lalrpop (hunk 1 above)
Comment (Marwes) [inline parser/src/grammar.lalrpop]: The epsilon production has never been more obvious, no? :)

**Pull request merged**